packages/core/src/modules/dids/domain/verificationMethod/Multikey.ts

```diff
@@ -0,0 +1,50 @@
+import type { VerificationMethod } from './VerificationMethod'
+
+import { Key } from '../../../../crypto/Key'
+import { AriesFrameworkError } from '../../../../error'
+
+export const VERIFICATION_METHOD_TYPE_MULTIKEY = 'Multikey'
+
+type GetMultikeyOptions = {
+  did: string
+  key: Key
+  verificationMethodId?: string
+}
+
+/**
+ * Get a Multikey verification method.
+ */
+export function getMultikey({ did, key, verificationMethodId }: GetMultikeyOptions) {
+  if (!verificationMethodId) {
+    verificationMethodId = `${did}#${key.fingerprint}`
+  }
+
+  return {
+    id: verificationMethodId,
+    type: VERIFICATION_METHOD_TYPE_MULTIKEY,
+    controller: did,
+    publicKeyMultibase: key.fingerprint,
+  }
+}
+
+/**
+ * Check whether a verification method is a Multikey verification method.
+ */
+export function isMultikey(
+  verificationMethod: VerificationMethod
+): verificationMethod is VerificationMethod & { type: 'Multikey' } {
+  return verificationMethod.type === VERIFICATION_METHOD_TYPE_MULTIKEY
+}
+
+/**
+ * Get a key from a Multikey verification method.
+ */
+export function getKeyFromMultikey(verificationMethod: VerificationMethod & { type: 'Multikey' }) {
+  if (!verificationMethod.publicKeyMultibase) {
+    throw new AriesFrameworkError(
+      `Missing publicKeyMultibase on verification method with type ${VERIFICATION_METHOD_TYPE_MULTIKEY}`
+    )
+  }
+
+  return Key.fromFingerprint(verificationMethod.publicKeyMultibase)
```

Comment:
fingerprint was supposed to give a fingerprint of the key, and it also is a multibase encoding. Might be good to add a toMultibase and fromMultibase options and also support e.g. base64 encoding as described in the di spec. But fine for now